lib/mpool/lib/mblock_file.c

```diff
@@ -997,6 +1037,65 @@ mblock_file_delete(struct mblock_file *mbfp, uint64_t *mbidv, int mbidc)
     return 0;
 }
 
+merr_t
+mblock_punch(struct mblock_file *mbfp, uint64_t mbid, off_t off, size_t len)
+{
+    struct mblock_oid_info mbinfo;
+    uint32_t block;
+    bool prealloced;
+    size_t wlen;
+    merr_t err;
+    char *addr;
+    int rc;
+
+    INVARIANT(mbfp);
+
+    if (!PAGE_ALIGNED(off) || !PAGE_ALIGNED(len))
```

Comment:
Seems like to me this should be checked at the mpool_mblock_punch() level and these should be assertions. Potentially even the next set of conditions on line 1058 too.

Reply:
hmm. The mblock layer (mblock.c) is not aware of the off/len requirements, mblock wlen etc. It is the mblock file layer (mblock_file.c) that is aware of all these and can enforce it.

Reply:
Makes sense then